plugins/pulp_rpm/plugins/importers/yum/pulp_solv.py

```diff
@@ -757,6 +710,9 @@ def add_repo_units(self, units, repo_id, installed=False):
         if installed:
             self._pool.installed = repo
 
+        # Need to call pool->addfileprovides(), pool->createwhatprovides() after loading new repo
+        self._finalized = False
```

Comment:
What is the situation when `self._finalized` can be `True` here?

Reply:
If it's already been "finalized" but then the user goes to add a new source/target repo.

The if you're using it sensibly it'll never happen, but the guard only took 2 seconds to write.